src/include/dhcp-server/dialogs.rb

```diff
@@ -47,39 +49,51 @@ def ReadDialog
     end
 
     # Write settings dialog
-    # @return `abort if aborted and `next otherwise
+    #
+    # @return [Symbol] :next whether configuration is saved successfully
+    #                  :back if user decided to change settings
+    #                  :abort otherwise
     def WriteDialog
-      Builtins.y2milestone("Running write dialog")
-      Wizard.RestoreHelp(Ops.get(@HELPS, "write", ""))
-      ret = DhcpServer.Write
-      if ret && restart_after_writing?
-        # Restart only if it's already running
-        DhcpServerUI.service.try_restart
-      end
-      # yes-no popup
-      if !ret &&
-          Popup.YesNo(
-            _("Saving the configuration failed. Change the settings?")
-          )
-        return :back
+      log.info("Running write dialog")
+
+      write_help = @HELPS.fetch("write") { "" }
+      Wizard.RestoreHelp(write_help)
+
+      return :next if write_settings
+      return :back if change_settings?
+
+      :abort
+    end
+
+    # Write DHCP server settings and save the service
+    #
+    # @return [Boolean] true if settings are saved successfully; false otherwise
+    def write_settings
+      if DhcpServer.Write
+        dhcp_service.save
```

Comment:
this looks wrong for me. It should be part of DhcpServer. So this service belongs to backend and not UI. Ideally DhcpServer holds service and see below.